static/openapi/faas-spec.json

```diff
@@ -0,0 +1,670 @@
+            "description": "Successfully retrieved the function names."
+          },
+          "404": {
+            "description": "Not found."
+          },
+          "500": {
+            "description": "Internal server error."
+          }
+        }
+      }
+    },
+    "/api/faas/v1/function/{functionName}/versions": {
+      "get": {
+        "tags": [
+          "Function"
+        ],
+        "summary": "Retrieve all versions",
+        "description": "Retrieve all versions for the specified key function name.",
+        "operationId": "handle_mm_versions",
+        "parameters": [
+          {
+            "name": "X-PhotonIQ-CustomerID",
+            "in": "header",
+            "description": "The customer ID",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "functionName",
+            "in": "path",
+            "description": "The function name",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the function versions."
+          },
+          "404": {
+            "description": "Not found."
+          },
+          "500": {
+            "description": "Internal server error."
+          }
+        }
+      }
+    },
+    "/api/faas/v1/execute/{functionName}": {
+      "post": {
+        "tags": [
+          "Function"
+        ],
+        "summary": "Execute the latest version of a function",
+        "description": "Execute the latest version of a function for the specified function name on body.",
+        "operationId": "handle_latest_mm_execute",
+        "parameters": [
+          {
+            "name": "X-PhotonIQ-CustomerID",
+            "in": "header",
+            "description": "The customer ID",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "functionName",
+            "in": "path",
+            "description": "The function name",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "requestBody": {
+          "description": "A valid JSON body is accepted",
+          "content": {
+            "application/json": {
+              "schema": {},
+              "example": {}
+            }
+          },
+          "required": true
+        },
+        "responses": {
+          "200": {
+            "description": "Successfully executed the function."
+          }
+        }
+      }
+    },
+    "/api/faas/v1/execute/{functionName}/{versionOrAlias}": {
+      "post": {
+        "tags": [
+          "Function"
+        ],
+        "summary": "Execute the function",
+        "description": "Execute the function for the specified function name on body.",
```

Comment:
I have no idea what this does.

Executes a function.  OK, what is this versionOrAlias thing?  How is this different from the above function?

Reply:
In one endpoint we can execute a function specifying its name+version or name+alias and for the other endpoint we can inform only its name, this last case will execute the latest version automatically

Reply:
This has to be explictly stated in the description of the API.

It looks like this API allows you to execute the function by either the version or by the alias.  Describe that in the API.